app/components/Form/ProjectFundingAgreementForm.tsx

```diff
@@ -0,0 +1,192 @@
+          formDataTableName: "funding_parameter"
+          filter: { operation: { notEqualTo: ARCHIVE } }
+        ) @connection(key: "connection_projectFundingAgreementFormChanges") {
+          __id
+          edges {
+            node {
+              id
+              rowId
+              newFormData
+              changeStatus
+            }
+          }
+        }
+      }
+    `,
+    props.projectRevision
+  );
+
+  const fundingAgreement =
+    projectRevision.projectFundingAgreementFormChanges.edges[0]?.node;
+
+  const addFundingAgreement = () => {
+    createFundingParameterFormChange({
+      variables: {
+        input: {
+          projectRevisionId: projectRevision.rowId,
+          formDataSchemaName: "cif",
+          formDataTableName: "funding_parameter",
+          jsonSchemaName: "funding_parameter",
+          operation: "CREATE",
+          newFormData: {
+            projectId: projectRevision.projectFormChange.formDataRecordId,
+          },
+        },
+        connections: [projectRevision.projectFundingAgreementFormChanges.__id],
+      },
+    });
+  };
+
+  const handleChange = ({ formData }) => {
+    // don't trigger a change if the form data is an empty object
+    if (Object.keys(formData).length === 0) return;
+
+    if (fundingAgreement) {
+      const updatedFormData = {
+        ...fundingAgreement.newFormData,
+        ...formData,
+      };
+      updateFundingParameterFormChange({
+        variables: {
+          input: {
+            id: fundingAgreement.id,
+            formChangePatch: {
+              newFormData: updatedFormData,
+              changeStatus: "pending",
+            },
+          },
+        },
+        optimisticResponse: {
+          updateFormChange: {
+            formChange: {
+              id: fundingAgreement.id,
+              newFormData: updatedFormData,
+              changeStatus: "pending",
+            },
+          },
+        },
+        debounceKey: fundingAgreement.id,
+      });
+    }
+  };
+
+  return (
+    <>
+      {projectRevision.projectFundingAgreementFormChanges.edges.length ===
+        0 && (
+        <Button onClick={addFundingAgreement} style={{ marginRight: "1rem" }}>
+          Add Funding Agreement
+        </Button>
+      )}
+      {projectRevision.projectFundingAgreementFormChanges.edges.length > 0 && (
+        <>
+          <header>
+            <h3>Project Funding Agreement</h3>
+            <UndoChangesButton formChangeIds={[fundingAgreement?.rowId]} />
+            <SavingIndicator
+              isSaved={
+                !isUpdatingFundingParameterFormChange &&
+                !isAddingFundingParameterFormChange
+              }
+            />
+          </header>
+
+          <FormBase
+            id="ProjectFundingAgreementForm"
+            validateOnMount={fundingAgreement?.changeStatus === "staged"}
+            idPrefix="ProjectFundingAgreementForm"
+            schema={createFundingAgreementSchema() as JSONSchema7}
```

Comment:
```suggestion
            schema={fundingAgreementSchema as JSONSchema7}
```
(if the createFundingAgreementSchema() function is removed)